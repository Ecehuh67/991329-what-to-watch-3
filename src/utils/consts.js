export const DEFAULT_GENRE = `All genres`;
export const DEFAULT_SHOWED_FILMS = 8;
