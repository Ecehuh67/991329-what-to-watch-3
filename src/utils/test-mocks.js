export const movies = [
  {
    title: `The Lord of the Rings`,
    image: `img/the-lord-of-the-rings.jpg`
  },
  {
    title: `The Ghostbusters`,
    image: `img/the-ghostbusters.jpg`
  },
  {
    title: `The Mortal Combat`,
    image: `img/the-mortal-combat.jpg`
  }
];
