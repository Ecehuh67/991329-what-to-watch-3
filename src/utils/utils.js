const AMOUT_SIMILAR_FILMS = 4;

export const getRandomNumber = (max) => {
  return 1 + Math.floor(Math.random() * max);
};

export const getSimilarFilms = (films, targetFilm) => {
  const index = films.map((it) => it.title).indexOf(targetFilm.title);
  const newList = [].concat(films.slice(0, index), films.slice(index + 1));

  return newList
    .slice()
    .filter((movie) => movie.genre === targetFilm.genre)
    .slice(0, AMOUT_SIMILAR_FILMS);
};
