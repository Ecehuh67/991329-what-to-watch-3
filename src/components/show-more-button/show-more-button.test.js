import ShowMoreButton from './show-more-button';
import {movies} from '../../utils/test-mocks';
import configureStore from "redux-mock-store";
import {Provider} from "react-redux";

const mockStore = configureStore([]);

describe(`Render ShowMoreButton correctly`, () => {
  it(`There is no showMoreButton`, () => {
    const store = mockStore({
      films: movies,
      showedFilms: 10,
    });

    const tree = renderer
      .create(
          <Provider store={store}>
            <ShowMoreButton
              onShowMoreButtonClick={() => {}}
              showedFilms={store.showedFilms}
              films={store.films}
            />
          </Provider>
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });
  it(`There is the showMoreButton`, () => {
    const store = mockStore({
      films: movies,
      showedFilms: 0,
    });

    const tree = renderer
      .create(
          <Provider store={store}>
            <ShowMoreButton
              onShowMoreButtonClick={() => {}}
              showedFilms={0}
              films={movies}
            />
          </Provider>
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });
});
