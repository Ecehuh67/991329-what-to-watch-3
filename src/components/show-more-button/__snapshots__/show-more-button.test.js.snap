// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render ShowMoreButton correctly There is no showMoreButton 1`] = `null`;

exports[`Render ShowMoreButton correctly There is the showMoreButton 1`] = `
<div
  className="catalog__more"
>
  <button
    className="catalog__button"
    onClick={[Function]}
    type="button"
  >
    Show more
  </button>
</div>
`;
